quil-rs/src/validation/identifier.rs

```diff
@@ -0,0 +1,34 @@
+//! Types and functions related to validating Quil identifiers
+use std::str::FromStr;
+
+use fancy_regex::Regex;
```

Comment:
`fancy_regex` supports negative look behind, which we use as part of `IDENTIFIER_REGEX` to ensure that an identifier doesn't end with a `-`, even though the middle characters of the identifier can contain it.

Reply:
Is lookbehind needed to enforce this?